From f27d6fc608e157b924a4d0e704b1dd76655bedbc Mon Sep 17 00:00:00 2001
From: "Brian C. Lane" <bcl@redhat.com>
Date: Mon, 9 Mar 2015 11:10:36 -0700
Subject: [PATCH] Reset environment to ensure consistent results (#1199434)

Set LANG, LC_ALL to C, set TZ to UTC and unset CDPATH and TERM.
---
 tests/t-lib.sh | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/tests/t-lib.sh b/tests/t-lib.sh
index d4411b5..4f68be4 100644
--- a/tests/t-lib.sh
+++ b/tests/t-lib.sh
@@ -15,6 +15,19 @@
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
+# For repeatability, reset the environment to known value.
+LANG=C
+LC_ALL=C
+TZ=UTC
+export LANG LC_ALL TZ
+
+# Protect ourselves from common misconfiguration to export
+# CDPATH into the environment
+unset CDPATH
+
+# Avoid spurious test failures due to buggy ncurses-5.6.
+unset TERM
+
 # Skip this test if the shell lacks support for functions.
 unset function_test
 eval 'function_test() { return 11; }; function_test'
-- 
2.1.0

